jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -63,16 +64,54 @@ impl Block {
     }
 
     /// The transactions contained in the block
-    pub fn transactions(&self, context: &Context) -> FieldResult<Vec<Transaction>> {
-        Ok(self
-            .get_explorer_block(&context.db)?
-            .transactions
-            .iter()
-            .map(|(id, _tx)| Transaction {
-                id: id.clone(),
-                in_block: self.hash.clone(),
-            })
-            .collect())
+    pub fn transactions(
+        &self,
+        first: Option<i32>,
+        last: Option<i32>,
+        before: Option<IndexCursor>,
+        after: Option<IndexCursor>,
```

Comment:
These four would be good to group as fields in a pagination parameter struct.

Reply:
Yeah, the resolver API being generic (the pagination param names are just a React convention, right?) does not mean it has to percolate into our internal Rust code organization.

Reply:
OTOH, if you say these functions are consumed by juniper and there is no way to change that, let it be.